src/main/scala/io/iohk/ethereum/jsonrpc/server/http/FaucetJsonRpcHttpServer.scala

```diff
@@ -0,0 +1,97 @@
+package io.iohk.ethereum.jsonrpc.server.http
+
+import java.security.SecureRandom
+import java.time.Clock
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{RemoteAddress, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Route, StandardRoute}
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives.cors
+import com.twitter.util.LruMap
+import io.iohk.ethereum.faucet.jsonrpc.FaucetJsonRpcController
+import io.iohk.ethereum.jsonrpc.server.controllers.JsonRpcBaseController
+import io.iohk.ethereum.jsonrpc.server.http.JsonRpcHttpServer.JsonRpcHttpServerConfig
+import io.iohk.ethereum.jsonrpc.{JsonRpcHealthChecker, JsonRpcRequest}
+import io.iohk.ethereum.utils.Logger
+import monix.execution.Scheduler.Implicits.global
+
+trait FaucetJsonRpcHttpServer extends JsonRpcHttpServer {
+
+  val minRequestInterval: Int
+
+  val latestTimestampCacheSize: Int
+
+  val latestRequestTimestamps: LruMap[RemoteAddress, Long]
+
+  val clock: Clock = Clock.systemUTC()
+
+  override def route: Route = cors(corsSettings) {
+    (pathEndOrSingleSlash & post) {
+      (extractClientIP & entity(as[JsonRpcRequest])) { (clientAddress: RemoteAddress, request: JsonRpcRequest) =>
```

Comment:
Is it enough to check IP address only in HTTP header, or maybe we shouldn't trust them?
Maybe we have to read `AttributeKeys.remoteAddress`? What do you think?

Reply:
HI! Thanks for the comment! I think you're right. I'll create a different task for it (since it might escape the scope of this one -> just bringing back old behavior)